dev-lang/rust/rust-1.29.0.recipe

```diff
@@ -0,0 +1,170 @@
+SUMMARY="Modern and safe systems programming language"
+DESCRIPTION="Rust is a systems programming language that runs blazingly fast, \
+prevents almost all crashes*, and eliminates data races."
+HOMEPAGE="https://www.rust-lang.org/"
+COPYRIGHT="2018 The Rust Project Developers"
+LICENSE="MIT"
+REVISION="1"
+cargoVersion="0.30.0"
+rlsVersion="0.130.0"
+rustfmtVersion="0.99.1"
+clippyVersion="0.0.212"
+SOURCE_URI="https://static.rust-lang.org/dist/rustc-$portVersion-src.tar.xz"
+CHECKSUM_SHA256="3943da98fb478a336ede7404e42ff76ef6ba4fc2b82012cfccd6b9fc4bd2c191"
+SOURCE_DIR="rustc-$portVersion-src"
+PATCHES="rust-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
```

Comment:
Suggestion:
```diff
 ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="?x86"
+fi
```

Reply:
Is this new?

Reply:
I haven't tested the recipe yet, but it seems **almost** OK.
To make it **100% valid** (for all possible combinations of primary vs secondary architectures) we need to apply the changes described in A **or** B **or** C.

**A)**
```diff
 ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="?x86"
+fi
```
(for `rust` and `rust_bin` of course.)

**B)**
```diff
 ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
 SECONDARY_ARCHITECTURES="?x86"
+
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+fi
```
and
```diff
	lib:libz$secondaryArchSuffix
	"
+if [ -n "$secondaryArchSuffix" -a -z "$commandSuffix" ]; then
+CONFLICTS="
+	rust
+	"
+fi
 
 BUILD_REQUIRES="
```
for `rust`.
For `rust_bin` we would probably require many more changes for `rust_bin_x86` to be valid on x86_64 hybrid.

**C)**
```diff
 ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
 SECONDARY_ARCHITECTURES="?x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
 
 PROVIDES="
 	rust$secondaryArchSuffix = $portVersion
-	cmd:rustc = $portVersion
+	cmd:rustc$commandSuffix = $portVersion
-	cmd:rustdoc = $portVersion
+	cmd:rustdoc$commandSuffix = $portVersion
-	cmd:rust_gdb = $portVersion
+	cmd:rust_gdb$commandSuffix = $portVersion
-	cmd:rust_lldb = $portVersion
+	cmd:rust_lldb$commandSuffix = $portVersion
 	cmd:cargo$secondaryArchSuffix = $cargoVersion
```
and more changes in `BUILD` inside a `if [ -n "$commandSuffix" ]; then` ... `fi` to move the `cmd:` runtimes from `$commandBinDir` to `$binDir`.
(Remark: we usually move the runtimes from `$binDir` to `$commandBinDir` inside a `if [ -z "$commandSuffix" ];` ... `fi` but here the runtimes get first installed in `$prefix/bin`, so we would need to move them to `$binDir` when we build `rust_x86` for x86_64 hybrid.)

## Question:

Do we want to have `rust_x86` (and `rust_bin_x86`) on x86_64 hybrid ? | Best solution
--- | ---
Yes | B
No | A

My guess is that we **don't** need `rust_x86` (nor `rust_bin_x86`) on x86_64; that's why I suggested solution A in my comment above.

BTW, regardless of which solution we choose between A, B or C, I just realized that we forgot to discuss about a possible change in `rust_bin`'s `PROVIDES`:
```diff
 PROVIDES="
-	rust_bin$secondaryArchSuffix = $portVersion
+	rust${secondaryArchSuffix}_bin = $portVersion
```
(IDK if this is desirable or not.)

Reply:
I think there is no need to have rust_x86 on x86_64. The use case we want to provide for is to be able to cross-compile x86 binaries from x86_64. Rust itself is very capable of cross-compiling for a different target. This of course needs to be tested, as it interacts with the local toolchain and the installed libraries.